app/src/test/java/org/simple/clinic/storage/files/AndroidFileStorageTest.kt

```diff
@@ -0,0 +1,42 @@
+package org.simple.clinic.storage.files
+
+import android.app.Application
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.File
+import java.io.IOException
+
+@RunWith(JUnitParamsRunner::class)
+class AndroidFileStorageTest {
+
+  @Test
+  @Parameters(method = "params for exceptions thrown when getting file")
+  fun `exceptions thrown when creating a file must be wrapped in the failure result`(cause: Throwable) {
```

Comment:
We wanted to do it, but this is implemented using extension functions (`bufferedReader()`, `bufferedWriter()`) on the `File` class which are static functions. We can't mock these using `Mockito` and the only way to change their behaviour in tests is using other tools like `Powermock`.

Since this is a very simple method, we opted not to go down the route of introducing another mocking library and forgoing testing these methods instead.